control-plane/src/health.rs

```diff
@@ -165,10 +180,12 @@ mod health_check_tests {
         assert_eq!(response.status(), 500);
         println!("deep response: {response:?}");
         let health_check_log = response_to_health_check_log(response).await;
-        assert!(matches!(
-            health_check_log.data_plane.status,
-            HealthCheckStatus::Err
-        ));
+
+        let dp_status = match health_check_log.data_plane {
+            HealthCheckVersion::V0(log) | HealthCheckVersion::V1(log) => log.status,
```

Comment:
Why are we matching against either of them? We know the expected version, it should be treated as an error if we incorrectly wrap the response.

Reply:
sorry ty this was a leftover - cherried from the "next" PR